packages/metal-state/test/Config.js

```diff
@@ -124,6 +124,34 @@ describe('Config', function() {
 		assert.ok(config.config.validator(['one']) instanceof Error);
 	});
 
+	it('should return config with "arrayOf" validator inheriting "shapeOf" and "oneOf" from "validators"', function(){
+		var shape = {
+			one: Config.bool().value(false),
+			two: Config.string(),
+			three: Config.oneOf([
+				'propOne',
+				'propTwo',
+				'propThree'
+			]).value('propOne'),
+			four: Config.number().required()
+		};
+		var configShape = Config.arrayOf(Config.shapeOf(shape));
+		assert.ok(core.isObject(configShape));
+		assert.ok(core.isFunction(configShape.config.validator));
+		assert.ok(configShape.config.validator({
```

Comment:
Why use `configShape.config.validator` not only needed to pass `configShape({...})`?

Reply:
I've based on other tests in config and discussing with @matuzalemsteles 
personally, i'll keep this.